test/it/group-list.js

```diff
@@ -32,6 +32,6 @@ describe('Group API tests', () => {
     expect(groupPresent).to.equal(true);
 
     // 3. Delete the group
-    await client.deleteGroup(createdGroup.id);
+    await utils.cleanup(client, null, createdGroup);
```

Comment:
Mocha supports the describe.only() option that will run only the specified test. And that is how the developer would debug one test. Due to this, we can't really have a single cleanup test that does the cleanup for all scenario tests. 
We still need to be able to cleanup individual test data. I do like using the `after` block for cleanup, but it's not always considered good practice.